utils/src/mem_size.rs

```diff
@@ -0,0 +1,66 @@
+use std::{collections::HashSet, sync::Arc};
+
+use parking_lot::RwLock;
+
+#[derive(Clone, Copy)]
+pub enum MemSize {
```

Comment:
Please add comments about intended use and where/when using this would be relevant.